rust_dev_preview/examples/rds/src/aurora.rs

```diff
@@ -0,0 +1,664 @@
+                } else {
+                    return Err(ScenarioError::new(
+                        "Could not create Cluster Parameter Group",
+                        &error,
+                    ));
+                }
+            }
+            _ => {
+                info!("Created Cluster Parameter Group");
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn set_instance_class(&mut self, instance_class: Option<String>) {
+        self.instance_class = instance_class;
+    }
+
+    pub fn set_login(&mut self, username: Option<String>, password: Option<SecretString>) {
+        self.username = username;
+        self.password = password;
+    }
+
+    pub async fn connection_string(&self) -> Result<String, ScenarioError> {
+        let cluster = self.get_cluster().await?;
+        let endpoint = cluster.endpoint().unwrap_or_default();
+        let port = cluster.port().unwrap_or_default();
+        let username = cluster.master_username().unwrap_or_default();
+        Ok(format!("mysql -h {endpoint} -P {port} -u {username} -p"))
+    }
+
+    pub async fn get_cluster(&self) -> Result<DbCluster, ScenarioError> {
+        let describe_db_clusters_output = self
+            .client
+            .describe_db_clusters()
+            .set_db_cluster_identifier(self.db_cluster_identifier.clone())
+            .send()
+            .await;
+        if let Err(err) = describe_db_clusters_output {
+            return Err(ScenarioError::new("Failed to get cluster", &err));
+        }
+
+        let db_cluster = describe_db_clusters_output
+            .unwrap()
+            .db_clusters
+            .and_then(|output| output.first().cloned());
+
+        db_cluster.ok_or_else(|| ScenarioError::with("Did not find the cluster"))
+    }
+
+    // Get the parameter group. rds.DescribeDbClusterParameterGroups
+    // Get parameters in the group. This is a long list so you will have to paginate. Find the auto_increment_offset and auto_increment_increment parameters (by ParameterName). rds.DescribeDbClusterParameters
+    // Parse the ParameterName, Description, and AllowedValues values and display them.
+    pub async fn cluster_parameters(&self) -> Result<Vec<AuroraScenarioParameter>, ScenarioError> {
+        let mut describe_db_cluster_parameters_paginator = self
+            .client
+            .describe_db_cluster_parameters()
+            .db_cluster_parameter_group_name(DB_CLUSTER_PARAMETER_GROUP_NAME)
+            .into_paginator()
+            .send();
+
+        let mut parameters: Vec<aws_sdk_rds::types::Parameter> = vec![];
+
+        loop {
+            let output = describe_db_cluster_parameters_paginator.next().await;
+            match output {
+                Some(output) => match output {
+                    Ok(output) => {
+                        for parameter in output.parameters() {
+                            parameters.push(parameter.clone());
```

Comment:
Stop pagination when both parameters are found?

Reply:
Unfortunately with the current pages implementation that doesn't really help. Refactored to a stream which would make that more opaque anyway. Lemme know your thoughts.